src/opm/output/eclipse/WriteRFT.cpp

```diff
@@ -256,26 +344,621 @@ namespace {
 
     void RFTRecord::addConnection(const ::Opm::UnitSystem&       usys,
                                   const ::Opm::Connection&       conn,
-                                  const ::Opm::data::Connection& xcon,
-                                  const double                   depth)
+                                  const ::Opm::data::Connection& xcon)
     {
-        this->i_.push_back(conn.getI() + 1);
-        this->j_.push_back(conn.getJ() + 1);
-        this->k_.push_back(conn.getK() + 1);
-
         using M = ::Opm::UnitSystem::measure;
         auto cvrt = [&usys](const M meas, const double x) -> float
         {
             return usys.from_si(meas, x);
         };
 
-        this->depth_.push_back(cvrt(M::length  , depth));
+        this->depth_.push_back(cvrt(M::length  , conn.depth()));
         this->press_.push_back(cvrt(M::pressure, xcon.cell_pressure));
 
         this->swat_.push_back(xcon.cell_saturation_water);
         this->sgas_.push_back(xcon.cell_saturation_gas);
+    }
 
-        this->host_.emplace_back();
+    // =======================================================================
+
+    class PLTFlowRate
+    {
+    public:
+        explicit PLTFlowRate(const std::size_t nconn = 0);
+
+        void addConnection(const ::Opm::UnitSystem&  usys,
+                           const ::Opm::data::Rates& rates);
+
+        const std::vector<float>& oil() const { return this->oil_; }
+        const std::vector<float>& gas() const { return this->gas_; }
+        const std::vector<float>& water() const { return this->water_; }
+
+    private:
+        std::vector<float> oil_{};
+        std::vector<float> gas_{};
+        std::vector<float> water_{};
+    };
+
+    PLTFlowRate::PLTFlowRate(const std::size_t nconn)
+    {
+        if (nconn == std::size_t{0}) {
+            return;
+        }
```

Comment:
> Is one really not allowed to call reserve with 0?

`vector<>::reserve(0)` is well defined, but I was trying to avoid doing extra work when it would not be needed.